src/Components/Web/src/Forms/InputFile/InputFileChangeEventArgs.cs

```diff
@@ -11,18 +11,48 @@ namespace Microsoft.AspNetCore.Components.Forms
     /// </summary>
     public sealed class InputFileChangeEventArgs : EventArgs
     {
+        private readonly IReadOnlyList<IBrowserFile> _files;
+
         /// <summary>
         /// Constructs a new <see cref="InputFileChangeEventArgs"/> instance.
         /// </summary>
         /// <param name="files">The list of <see cref="IBrowserFile"/>.</param>
         public InputFileChangeEventArgs(IReadOnlyList<IBrowserFile> files)
         {
-            Files = files;
+            _files = files ?? throw new ArgumentNullException(nameof(files));
         }
 
         /// <summary>
-        /// Gets the file entries list.
+        /// Gets the number of supplied files.
+        /// </summary>
+        public int FileCount => _files.Count;
+
+        /// <summary>
+        /// Gets the supplied file, or null if no file was supplied. Note that if the input
+        /// accepts multiple files, then instead of reading this property, you should call
+        /// <see cref="AcceptMultipleFiles(int)"/>.
         /// </summary>
-        public IReadOnlyList<IBrowserFile> Files { get; init; }
+        public IBrowserFile? File => _files.Count switch
+        {
+            0 => null,
```

Comment:
Can this really happen?

Reply:
I've been trying to work that out for the last little while, and it seems like no. I *thought* the user might be able to do some obscure gesture like focusing and pressing delete to clear the file input, but no gesture I've tried so far has that effect. The HTML spec doesn't seem to address this question directly.

Since there's no indication from my research that it's possible, I'll change this case to throw on the grounds that it's not meant to be possible. If somehow it does become possible in the future it would be awkward for us though - either we'd need a breaking change, or our API here would be kind of messy (you'd have to check the file count before accessing the `File` property).